### DISCUSSION TITLE
Concatenation of details

### DISCUSSION THREAD

**Discussion by Udaberrico** [php-edifact/edifact#143] [General]
Hello everyone, 

I have a problem here, and I don't know if it is the expected behavior or if it's actually not right. 

I have a part of a message in the format D99B. According to the documentation, partyName appears several times in succession. These are then combined into one word without a separator.

I have traced the whole thing back to the following PR https://github.com/php-edifact/edifact/pull/133

Is the concatenation the right way? In my example, I get something like `["partyName" => [
      "partyName" => "Company Ltd.FirstNameLastname"]]
`.
Now as this information is quite important because its the name of the invoice receiving company, shouldn't it be something like `["partyName" => [
      "partyName" => ["Company Ltd.", "FirstName", "Lastname"]]]
`
This would make it easier to process it further.

**Comment by sabas** [comment]
Are you referring to the Analyser behaviour or does also Interpreter do it in the same way?

Cc @jnmrclmbsjse
